spoc/io.py

```diff
@@ -73,12 +104,149 @@ def load_chromosome_sizes(path: str):
         )
 
     @staticmethod
-    def load_pixels(path: str):
-        """Loads pixels"""
-        # TODO:make this conform with new file format
-        raise NotImplementedError
+    def _load_metadata(path: str):
+        """Load metadata"""
+        metadata_path = Path(path) / "metadata.json"
+        if metadata_path.exists():
+            with open(metadata_path, "r", encoding="UTF-8") as f:
+                metadata = json.load(f)
+        else:
+            raise ValueError(f"Metadata file not found at {metadata_path}")
+        return metadata
+
+    @staticmethod
+    def list_pixels(path: str) -> List[PixelParameters]:
+        """List available pixels
+
+        Args:
+            path (str): Path to the pixel data.
+
+        Returns:
+            List[PixelParameters]: List of PixelParameters objects.
+        """
+        # read metadata.json
+        metadata = FileManager._load_metadata(path)
+        # instantiate pixel parameters
+        pixels = [PixelParameters(**params) for params in metadata.values()]
+        return pixels
+
+    @staticmethod
+    def list_contacts(path: str) -> List[ContactsParameters]:
+        """List available contacts"""
+        # read metadata.json
+        metadata = FileManager._load_metadata(path)
+        # instantiate pixel parameters
+        contacts = [ContactsParameters(**params) for params in metadata.values()]
+        return contacts
+
+    def load_pixels(
+        self, path: str, global_parameters: PixelParameters, load_dataframe: bool = True
+    ) -> Pixels:
+        """Loads specific pixels instance based on global parameters.
+        load_dataframe specifies whether the dataframe should be loaded, or whether pixels
+         should be instantiated based on the path alone.
+
+        Args:
+            path (str): Path to the pixel data.
+            global_parameters (PixelParameters): Global parameters.
+            load_dataframe (bool, optional): Whether to load the dataframe. Defaults to True.
+
+        Returns:
+            Pixels: Pixels object containing the pixel data.
+
+        """
+        metadata = self._load_metadata(path)
+        # find matching pixels
+        for pixel_path, value in metadata.items():
+            param_value = PixelParameters(**value)
+            if param_value == global_parameters:
+                selected_value = param_value
+                break
+        else:
+            raise ValueError(f"No matching pixels found for {global_parameters}")
+        # rewrite path to contain parent folder
+        pixel_path = Path(path) / pixel_path
+        if load_dataframe:
+            df = self._parquet_reader_func(pixel_path)
+        else:
+            df = pixel_path
+        return Pixels(df, **selected_value.dict())
+
+    def load_contacts(
```

Comment:
`load_contacts` is almost identical to `load_pixels`. Should we smhw unify duplicated code into `_load_table`, that is called inside these two methods?

Reply:
No, not for now.